core/complexStructures/processedMiniBlocks.go

```diff
@@ -0,0 +1,132 @@
+package complexStructures
+
+import (
+	"sync"
+
+	"github.com/ElrondNetwork/elrond-go/logger"
+	"github.com/ElrondNetwork/elrond-go/process/block/bootstrapStorage"
+)
+
+var log = logger.GetOrCreate("core/complexStructures")
+
+// ProcessedMiniBlocks is used to store all processed min blocks hashes for e meta block hash
+type ProcessedMiniBlocks struct {
+	processedMiniBlocks    map[string]map[string]struct{}
+	mutProcessedMiniBlocks sync.RWMutex
+}
+
+// NewProcessedMiniBlocks will create a complex type of processedMiniBlocks
+func NewProcessedMiniBlocks() *ProcessedMiniBlocks {
+	return &ProcessedMiniBlocks{
+		processedMiniBlocks: make(map[string]map[string]struct{}),
+	}
+}
+
+// AddMiniBlockHash will add a miniblock hash
+func (pmb *ProcessedMiniBlocks) AddMiniBlockHash(metaBlockHash []byte, miniBlockHash []byte) {
+	pmb.mutProcessedMiniBlocks.Lock()
+	defer pmb.mutProcessedMiniBlocks.Unlock()
+
+	miniBlocksProcessed, ok := pmb.processedMiniBlocks[string(metaBlockHash)]
+	if !ok {
+		miniBlocksProcessed := make(map[string]struct{})
+		miniBlocksProcessed[string(miniBlockHash)] = struct{}{}
+		pmb.processedMiniBlocks[string(metaBlockHash)] = miniBlocksProcessed
+
+		return
+	}
+
+	miniBlocksProcessed[string(miniBlockHash)] = struct{}{}
+}
+
+// RemoveMetaBlockHash will remove a meta block hash
+func (pmb *ProcessedMiniBlocks) RemoveMetaBlockHash(metaBlockHash []byte) {
+	pmb.mutProcessedMiniBlocks.Lock()
+	delete(pmb.processedMiniBlocks, string(metaBlockHash))
+	pmb.mutProcessedMiniBlocks.Unlock()
+}
+
+// RemoveMiniBlockHash will remove a mini block hash
+func (pmb *ProcessedMiniBlocks) RemoveMiniBlockHash(miniBlockHash []byte) {
+	pmb.mutProcessedMiniBlocks.Lock()
+	for metaHash, miniBlocksProcessed := range pmb.processedMiniBlocks {
+		_, isProcessed := miniBlocksProcessed[string(miniBlockHash)]
+		if isProcessed {
+			delete(miniBlocksProcessed, string(miniBlockHash))
+		}
+
+		if len(miniBlocksProcessed) == 0 {
+			delete(pmb.processedMiniBlocks, metaHash)
+		}
+	}
+	pmb.mutProcessedMiniBlocks.Unlock()
+}
+
+// GetProcessedMiniBlocksHashes will return all processed miniblocks for a metablock
+func (pmb *ProcessedMiniBlocks) GetProcessedMiniBlocksHashes(metaBlockHash []byte) map[string]struct{} {
+	pmb.mutProcessedMiniBlocks.RLock()
+	processedMiniBlocksHashes := pmb.processedMiniBlocks[string(metaBlockHash)]
+	pmb.mutProcessedMiniBlocks.RUnlock()
+
+	return processedMiniBlocksHashes
+}
+
+// IsMiniBlockProcessed will return true if a mini block is processed
+func (pmb *ProcessedMiniBlocks) IsMiniBlockProcessed(metaBlockHash []byte, miniBlockHash []byte) bool {
```

Comment:
use `string` as parameters' type because where you use these methods you cast the parameters as `[]byte` and in this structure you cast them back to `string`

Reply:
done.